indent/handlebars.vim

```diff
@@ -84,32 +81,35 @@ function! GetHandlebarsIndent(...)
 
   " indent after block {{#block
   if prevLine =~# '\v\s*\{\{\#.*\s*'
-    let s:in_block_expr = 1
     let ind = ind + sw
   endif
   " but not if the block ends on the same line
   if prevLine =~# '\v\s*\{\{\#(.+)(\s+|\}\}).*\{\{\/\1'
-    let s:in_block_expr = 0
     let ind = ind - sw
   endif
   " unindent after block close {{/block}}
   if currentLine =~# '\v^\s*\{\{\/\S*\}\}\s*'
-    let s:in_block_expr = 0
     let ind = ind - sw
   endif
   " indent after component block {{a-component
   if prevLine =~# '\v\s*\{\{\w'
-    let s:in_block_expr = 0
     let ind = ind + sw
   endif
   " but not if the component block ends on the same line
   if prevLine =~# '\v\s*\{\{\w(.+)\}\}'
     let ind = ind - sw
   endif
   " unindent }} lines, and following lines if not inside a block expression
+  let savedPos = getpos('.')
   if currentLine =~# '\v^\s*\}\}\s*$' || (currentLine !~# '\v^\s*\{\{\/' && prevLine =~# '\v^\s*[^\{\} \t]+\}\}\s*$')
-    if !s:in_block_expr
-      let ind = ind - sw
+    let closingLnum = search('}}\s*$', 'Wbc', lnum)
```

Comment:
Would it be better/possible to turn this whole section into a separate function, something like `IsInBlock`, so it can be used elsewhere if needed?

Reply:
I would rather have at least two places where it's used before extracting, so it cant be extracted in a way that works in both places, rather than guessing what to pull out and what to leave as a parameter.

I'm assuming you're thinking of this same pattern being used in my other PR? If you merge them both, you can open an issue for me, and I'll see about extracting the common code. Otherwise I need to merge the branches locally and I'll still need to make another commit after to the other PR.